dynosaur/tests/pass/basic.stdout

```diff
@@ -7,13 +7,17 @@ use dynosaur::dynosaur;
 
 trait MyTrait {
     type Item;
-    async fn foo(&self)
+    async fn foo()
     -> Self::Item;
 }
 trait ErasedMyTrait {
     type Item;
-    fn foo(&self)
-    -> ::core::pin::Pin<Box<dyn ::core::future::Future<Output = Self::Item>>>;
+    fn foo<'async_trait>()
+    ->
+        ::core::pin::Pin<Box<dyn ::core::future::Future<Output = Self::Item> +
+        'async_trait>>
+    where
+    Self: 'async_trait;
```

Comment:
Unsure if we should be adding this lifetime on a method that doesn't take self as a parameter.

Reply:
There's Self in the `Output =` part. Anyway, you're right in the sense that I'm not correctly checking if the function has self or not. Addressed that and added a new test.